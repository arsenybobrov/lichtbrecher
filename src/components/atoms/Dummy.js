import React, { useContext, useState, useEffect } from 'react';
import styled, { ThemeContext } from 'styled-components';
import { lighten } from 'polished';

const Title = styled.h1`
  color: ${(props) => (lighten(0.5, props.theme.colors.primary))};
  background-color: ${(props) => props.active ? 'red' : 'transparent'};
`;

const Dummy = ({ globalConfig, data }) => {
  const theme = useContext(ThemeContext);
  const [mounted, mount] = useState(false);
  const [clicked, toggleClick] = useState(mounted);

  useEffect(() => {
    setTimeout(() => {
      mount(true);
    }, 3000);
  }, []);

  return (
    <>
      {globalConfig && <Title active={clicked}>Global: {globalConfig.data.display_name[0].text}</Title>}
      {data && <Title active={clicked}>Data: {data.data.display_name[0].text}</Title>}
      {mounted && <button onClick={() => toggleClick(!clicked)}>Click me, my color is: {theme.colors.primary}</button>}
    </>
  );
};

export default Dummy;
