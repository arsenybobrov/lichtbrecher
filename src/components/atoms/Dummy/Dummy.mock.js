const DummyMock = {
  globalConfig: {
    data: {
      display_name: [
        { text: 'Dummy text' },
      ],
    },
  },
  data: {
    data: {
      display_name: [
        { text: 'Dummy text' },
      ],
    },
  },
};

export default DummyMock;
