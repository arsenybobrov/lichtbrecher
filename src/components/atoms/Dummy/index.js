import React, { useContext, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import get from 'lodash/get';
import { RichText } from 'prismic-reactjs';
import styled, { ThemeContext } from 'styled-components';
import { lighten } from 'polished';
import { GlobalConfigContext } from '../../../contexts/globalConfig';
import { BreakpointContext } from '../../../contexts/breakpoint';
import mq from '../../../styles/mediaQueries';
import { hrefResolver, linkResolver } from '../../../../api/prismic/linkResolver';

const Title = styled.h1`
  color: ${(props) => (lighten(0.5, props.theme.colors.primary))};
  background-color: ${(props) => (props.active ? 'red' : 'transparent')};

  @media (${mq.mdDown}) {
    background-color: black;
    color: white;
  }
`;

const Dummy = ({ data }) => {
  const theme = useContext(ThemeContext);
  const content = useContext(GlobalConfigContext);
  const breakpoint = useContext(BreakpointContext);
  const [mounted, mount] = useState(false);
  const [clicked, toggleClick] = useState(mounted);
  const richtText = RichText.render(data.data.text, linkResolver);

  useEffect(() => {
    setTimeout(() => {
      mount(true);
    }, 3000);
  }, []);

  return (
    <>
      {
        content &&
        <Title active={clicked}>Global: {content.data.display_name[0].text}</Title>
      }
      {
        data &&
        <>
          <Title active={clicked}>Page: {data.data.display_name[0].text}</Title>
          <Title active={clicked}>Text: {richtText}</Title>
        </>
      }
      <p>BREAKPOINT IS: {breakpoint}</p>
      {
        mounted &&
        <button type="button" onClick={() => toggleClick(!clicked)}>
          Click me, my text color is: {theme.colors.primary}
        </button>
      }
      <br />
      <br />
      {
        get(data, 'data.url', null) &&
        <Link as={hrefResolver(get(data, 'data.url', null))} href="[page]">
          <a>{get(data, 'data.linktext', 'default text')}</a>
        </Link>
      }
    </>
  );
};

Dummy.propTypes = {
  data: PropTypes.object,
};

export default Dummy;
