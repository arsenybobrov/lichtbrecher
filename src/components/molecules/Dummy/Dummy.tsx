import React from 'react';
import uniqueId from 'lodash/uniqueId';
import DummyHeadline, { HeadlineProps } from '../../atoms/DummyHeadline/DummyHeadline';
import DummyRichText, { RichTextElementType } from '../../atoms/DummyRichText/DummyRichText';


interface PrismicDummyContent {
 headline: Array<HeadlineProps>;
 richtext: Array<RichTextElementType>;
}

interface DummyProps {
  primary: PrismicDummyContent;
}

const Dummy: React.FC<DummyProps> = ({ primary }) => {
  const { headline, richtext } = primary;
  return (
    <>
      {headline.map((headlineProps) => <DummyHeadline {...headlineProps} key={uniqueId('headline')} />)}
      <DummyRichText content={richtext} />
    </>
  );
};

export default Dummy;
