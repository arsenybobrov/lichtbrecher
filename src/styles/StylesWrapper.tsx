import React from 'react';
import { ThemeProvider } from 'styled-components';
import BasicCSS from './css/BasicCSS';
import FontsCSS from './css/FontsCSS';
import defaultTheme from '../themes/default';

const StylesWrapper: React.FC = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>
    <BasicCSS />
    <FontsCSS />
    {children}
  </ThemeProvider>
);

export default StylesWrapper;
