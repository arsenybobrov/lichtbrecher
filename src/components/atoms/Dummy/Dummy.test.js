import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Dummy from './index';
import DummyMock from './Dummy.mock';
import StylesWrapper from '../../../styles/Wrapper';

describe('Dummy', () => {
  it.each`
    name                                | mock
    ${'description of expected result'} | ${DummyMock}
  `('$name', ({ mock }) => {
  const tree = renderer
    .create(
      <StylesWrapper>
        <Dummy {...mock} />
      </StylesWrapper>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
});
