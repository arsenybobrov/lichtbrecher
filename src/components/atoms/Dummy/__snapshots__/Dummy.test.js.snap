// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dummy description of expected result 1`] = `
Array [
  .c0 {
  color: #808080;
  background-color: transparent;
}

@media (max-width:991px) {
  .c0 {
    background-color: black;
    color: white;
  }
}

<h1
    className="c0"
  >
    Global: 
    Dummy text
  </h1>,
  .c0 {
  color: #808080;
  background-color: transparent;
}

@media (max-width:991px) {
  .c0 {
    background-color: black;
    color: white;
  }
}

<h1
    className="c0"
  >
    Page: 
    Dummy text
  </h1>,
  .c0 {
  color: #808080;
  background-color: transparent;
}

@media (max-width:991px) {
  .c0 {
    background-color: black;
    color: white;
  }
}

<h1
    className="c0"
  >
    Text: 
  </h1>,
  <p>
    BREAKPOINT IS: 
    xxl
  </p>,
  <br />,
  <br />,
]
`;
